### DISCUSSION TITLE
Stunningly good vocal and music separation

### DISCUSSION THREAD

**Discussion by petersampsonaudacity** [intel/openvino-plugins-ai-audacity#311] [General]
Hi @RyanMetcalfeInt8 

Previously I only did some early QA testing of the OpenVINO plugins for Audacity - but today I tried using it for real for vocal separation and the results were stunningly good.

I have long had the CD version of Emmylou Harris' Roses In The Snow album - this has a cover of Paul Simon's The Boxer as a bonus track.

This track has always troubled. annoyed me, as though Emmylou's singing is wonderful the instrumental backing is absolutely awful, far too regular and metronomic making the whole track sounds as though it just plods along with no excitement or insight.

A few years ago I recall trying vocal separation on this track with Audacity's built-in tool for the job, but the results were very disappointing, so I abandoned hope.

Today though I thought: let's try the OpenVINO tool for this job in Audacity - and the results were absolutely stunningly good with total separation of Emmylou's vocals from the backing "music" - the upshot being that I now have _(with a little editing out of unnecessary silences previously filled by the backing)_ a lovely acapella Emmylou version of the song.

So many thanks to you and the other folks at OpenVINO Ryan 

Cheers,
Peter

**Comment by RyanMetcalfeInt8** [comment]
Hi @petersampsonaudacity!

Thank you so much for the positive feedback! Thrilled that you were able to get some good results from these plugins for Music Separation. 'The Boxer' is one of my favorite songs -- I'll have to seek out this version to have a listen.

And just as a heads up -- I recently discovered another AI model which produces better overall results than the one we are currently using. I'd like to eventually add that as an option, so I'll let you know when it becomes available to test with. It would be interesting to see if you get even better results with that version.

Cheers!
Ryan

**Reply by MarcoRavich** [reply]
> And just as a heads up -- I recently discovered another AI model which produces better overall results than the one we are currently using. I'd like to eventually add that as an option, so I'll let you know when it becomes available to test with. It would be interesting to see if you get even better results with that version.

Nice, anyway it would be extremely useful to have a (audio-buses oriented ?) stem separator too like - ehm - this:
![iZOTOPE RX 11 - Music rebalance](https://github.com/user-attachments/assets/ff7e175b-8b90-4328-9fd3-0ed73ef31b2d)

...as always check out [Separating](https://github.com/FORARTfe/HyMPS/blob/main/Audio/AI-based.md#separating-) in the [AI-based page](https://github.com/FORARTfe/HyMPS/blob/main/Audio/AI-based.md#--) of the [AUDIO section](https://github.com/FORARTfe/HyMPS#-) of the [HyMPS project](https://forart.it/HyMPS/)'s resource collection.

**Reply by RyanMetcalfeInt8** [reply]
Cool -- thanks for the recommendation. If there is a known open source (pre-trained) model / surrounding algorithm that produces good results, that would be most helpful.

By the way, the AI model that I mention above (which produces higher quality vocals) is from this project:
https://github.com/KimberleyJensen/Mel-Band-Roformer-Vocal-Model

You may want to add that to your Separators page. I might've missed it, but I didn't spot it there.

**Reply by MarcoRavich** [reply]
> By the way, the AI model that I mention above (which produces higher quality vocals) is from this project: https://github.com/KimberleyJensen/Mel-Band-Roformer-Vocal-Model
> 
There's something similar for stem separation by @lucidrains: [BS-RoFormer](https://github.com/lucidrains/BS-RoFormer#readme) - _Implementation of [Band Split Roformer](https://arxiv.org/abs/2309.02612), SOTA Attention network for music source separation out of ByteDance AI Labs. They beat the previous first place by a large margin. The technique uses axial attention across frequency (hence multi-band) and time. They also have experiments to show that rotary positional encoding led to a huge improvement over learned absolute positions._

Anyway, as you probably already know, the "defacto" standard in stem separation is [Spleeter](https://github.com/deezer/spleeter#readme) by @deezer.

**Reply by Macktek** [reply]
I just wanted to chime in and say Thank you for making this! I was able to use it to remove excessively loud music from an Instructional Video. The results are phenomenal. I would have liked to have started a new Discussion, but I don't see any way to do that, so, I am replying here. 
GREAT WORK!

**Comment by petersampsonaudacity** [comment]
Hi @RyanMetcalfeInt8 

That track comes from Emmylou Harris's Roses In The Snow album.

For copyright reasons I can't post you a link on here to the source file - _but do keep an eye on your PM's on the Audacity Forum shortly_

The other great cover I have of the Boxer is by the great Dobro player Jerry Douglas - I don't know which album that's on as I "borrowed" it off-air using Audacity.  But it's well worth seeking out.

Cheers,
Peter.
